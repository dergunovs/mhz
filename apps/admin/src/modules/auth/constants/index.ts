export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
