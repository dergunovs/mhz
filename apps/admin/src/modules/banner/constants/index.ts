export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
