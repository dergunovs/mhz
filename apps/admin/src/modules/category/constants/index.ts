export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
