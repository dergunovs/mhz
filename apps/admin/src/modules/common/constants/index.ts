export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
