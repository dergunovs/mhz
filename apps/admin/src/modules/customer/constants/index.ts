export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
