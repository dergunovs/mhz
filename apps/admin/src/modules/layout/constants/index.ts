import { INavItem } from '@/layout/interface';

export const NAV_ITEMS: INavItem[] = [
  { _id: 0, url: '/', title: 'Главная' },
  { _id: 1, url: '/users', title: 'Пользователи' },
  { _id: 2, url: '/shipments', title: 'Доставка' },
  { _id: 3, url: '/orders', title: 'Заказы' },
  { _id: 4, url: '/categories', title: 'Категории' },
  { _id: 5, url: '/products', title: 'Товары' },
  { _id: 5, url: '/manufacturers', title: 'Производители' },
];
