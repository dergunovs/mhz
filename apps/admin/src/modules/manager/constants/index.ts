export const URL_MANAGER = '/managers';
