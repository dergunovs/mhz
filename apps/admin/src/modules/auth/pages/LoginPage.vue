<template>
  <div :class="$style.container">
    <LoginForm />
  </div>
</template>

<script setup lang="ts">
import { useRouter } from 'vue-router';

import LoginForm from '@/auth/components/LoginForm.vue';

import { isAuth } from '@/auth/composables';
import { URL_MAIN } from '@/common/constants';

const router = useRouter();

if (isAuth.value) router.push(URL_MAIN);
</script>

<style module lang="scss">
.container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
}
</style>
