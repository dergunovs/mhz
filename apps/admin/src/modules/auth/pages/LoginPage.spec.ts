import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import * as helpers from 'mhz-helpers';

import LoginPage from './LoginPage.vue';

import { wrapperFactory } from '@/common/test';
import { URL_MAIN } from '@/common/constants';

const spyRedirectIfAuth = vi.fn();
const spyAuth = vi.spyOn(helpers, 'useAuth').mockReturnValue({ auth: vi.fn(), redirectIfAuth: spyRedirectIfAuth });

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(LoginPage, {});
});

enableAutoUnmount(afterEach);

describe('LoginPage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(LoginPage)).toBeTruthy();
  });

  it('redirects to main page if authorized', async () => {
    expect(spyAuth).toBeCalledTimes(1);
    expect(spyRedirectIfAuth).toBeCalledTimes(1);

    expect(spyRedirectIfAuth).toBeCalledWith(URL_MAIN);
  });
});
