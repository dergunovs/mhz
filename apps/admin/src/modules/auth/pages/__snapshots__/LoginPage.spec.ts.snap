// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > matches snapshot 1`] = `
"<div>
  <login-form-stub></login-form-stub>
</div>"
`;
